README.md

```diff
@@ -1 +1,29 @@
 # Valigator
+
+Valigator is a lazy validation builder library. It allows for the reuse of validation across your code in an easy-to-use manner. 
+
+Usually, when doing validation, people end up writing validation logic in the constructor which is okay but you don't really end up being able to easily share that validation functionality across classes. 
+
+The idea is to somewhat decouple validation from your classes. So you perform validation based on a set of criteria. If you fulfill the criteria you should be able to create your Model based on that data. 
+
+The validator is a builder that takes validation functions to execute. You pass in the property, whether or not its optional and the list of validation functions you would like to run on the property
+
+![image](https://cloud.githubusercontent.com/assets/14280155/11577118/5c5d2b42-99e9-11e5-868f-d95a3ea55352.png)
```

Comment:
Also in the picture we instantiate it as Validator,  we should probably switch that to be Valigator? not sure if that is appropriate lol